pennylane/transforms/tape_expand.py

```diff
@@ -350,6 +350,80 @@ def custom_decomp_expand(self, circuit, max_expansion=decomp_depth):
     return custom_decomp_expand
 
 
+def create_decomp_preprocessing(custom_decomps, dev, decomp_depth=10):
```

Comment:
Is this intended to be user facing?

Reply:
I think no? But is `create_decomp_expand_function` user facing? This is that but for the new device API.

Reply:
`create_decomp_expand_function` is not in the docs :+1:

Reply:
Can we have `create_decomp_preprocessing` with a leading underscore?